crates/optimism/op-chain-registry/src/chain-registry.rs

```diff
@@ -0,0 +1,128 @@
+//! Directory Manager downloads and manages files from the op-superchain-registry
+
+use anyhow::{Context, Result};
+use serde_json::Value;
+use std::{
+    fs,
+    path::{Path, PathBuf},
+};
+use tokio::fs as tokio_fs;
+use zstd::{dict::DecoderDictionary, stream::read::Decoder};
+
+/// Directory manager that handles caching and downloading of genesis files
+#[derive(Debug)]
+#[allow(dead_code)]
+pub struct DirectoryManager {
+    base_path: PathBuf,
+}
+
+#[allow(dead_code)]
+impl DirectoryManager {
+    const DICT_URL: &'static str = "https://raw.githubusercontent.com/ethereum-optimism/superchain-registry/main/superchain/extra/dictionary";
+    const GENESIS_BASE_URL: &'static str = "https://raw.githubusercontent.com/ethereum-optimism/superchain-registry/main/superchain/extra/genesis";
+
+    /// Create a new registry manager with the given base path
+    pub(crate) fn new(base_path: PathBuf) -> Result<Self> {
+        fs::create_dir_all(&base_path)?;
+        Ok(Self { base_path })
+    }
+
+    /// Get the path to the dictionary file
+    fn dictionary_path(&self) -> PathBuf {
+        self.base_path.join("dictionary")
+    }
+
+    /// Get the path to a genesis file
+    fn genesis_path(&self, network_type: &str, network: &str) -> PathBuf {
+        self.base_path.join(network_type).join(format!("{}.json.zst", network))
+    }
+
+    /// Check if a file exists at the given path
+    async fn file_exists(&self, path: &Path) -> bool {
+        path.exists()
+    }
+
+    /// Read file from the given path
+    async fn read_file(&self, path: &Path) -> Result<Vec<u8>> {
+        tokio_fs::read(path).await.context("Failed to read file")
+    }
+
+    /// Save data to the given path
+    async fn save_file(&self, path: &Path, data: &[u8]) -> Result<()> {
+        if let Some(parent) = path.parent() {
+            fs::create_dir_all(parent)?;
+        }
+        tokio_fs::write(path, data).await.context("Failed to write file")
+    }
+
+    /// Download a file from the given URL
+    async fn download_file(&self, url: &str, path: &Path) -> Result<Vec<u8>> {
```

Comment:
the way we need to use this will be via blocking IO

so we don't gain anything from these functions being async, so I'd rather have them sync